tools/replay/parse_dump_resources_cli.cpp

```diff
@@ -191,20 +158,48 @@ static bool CheckIndicesForErrors(const gfxrecon::decode::VulkanReplayOptions& v
         return true;
     }
 
-    if (vulkan_replay_options.Draw_Indices.size() && !vulkan_replay_options.RenderPass_Indices.size())
+    // Make sure each Draw has corresponding RenderPass
+    if (vulkan_replay_options.Draw_Indices.size() != vulkan_replay_options.RenderPass_Indices.size())
     {
         GFXRECON_LOG_ERROR("ERROR - incomplete --dump-resources parameters");
-        GFXRECON_LOG_ERROR("DrawCall indices must always be acompanied by render pass indices");
+        GFXRECON_LOG_ERROR("Draw indices must always be accompanied by RenderPass indices");
         return true;
     }
+    for (int i = 0; i < vulkan_replay_options.Draw_Indices.size(); i++)
+    {
+        if (vulkan_replay_options.Draw_Indices[i].size() != vulkan_replay_options.RenderPass_Indices[i].size())
```

Comment:
This check is wrong. It will print a warning for valid input.
Perhaps you want to do
```
if (!vulkan_replay_options.Draw_Indices[i].empty() && vulkan_replay_options.RenderPass_Indices[i].empty())
```

Reply:
Not exactly.
When specifying a draw call for dumping you need to specify the indices of BeginRenderPass and EndRenderPass that include that draw call. If you want to dump multiple draw calls from the same render pass you don't have to specify the indices for Begin/End again. Like in this example

```
    "Draw": [
        [
            473,
            474,
            475,
            476,
            477,
            478,
            479,
            480,
            481,
            482,
            483,
            484,
            485,
            486
        ]
    ],
    "RenderPass": [
        [
            [
                466,
                487
            ]
        ]
    ],
```
In the example above it will be
`vulkan_replay_options.Draw_Indices[0].size() == 14` and `vulkan_replay_options.RenderPass_Indices[0].size() == 1`